### PR TITLE
Use diff-patch-match for TextDifferencer

### PR DESCRIPTION
**Pull request by oferb**

### CODE REVIEW

File: common/TextDifferencer.java [hunk 1]
```diff
@@ -29,7 +29,6 @@
 import javax.swing.text.Segment;
 import java.util.LinkedList;
```
Comment (val-fed) [inline common/TextDifferencer.java]: Unused imports:
- import java.util.ArrayList;
- import java.util.Collections;
- import java.util.List;
- import java.util.stream.Collectors;
- import java.util.stream.IntStream;
- import javax.swing.text.Segment;

File: common/tests/TextDifferencerTool.java [hunk 2]
```diff
@@ -32,7 +32,6 @@
 import name.fraser.neil.plaintext.DiffMatchPatch.Operation;
 import java.util.LinkedList;
```
Comment (val-fed) [inline common/tests/TextDifferencerTool.java]: Unused imports:
- import com.google.startupos.common.TextChange.Type;
- import com.google.startupos.name.fraser.neil.plaintext.DiffMatchPatch;
- import com.google.startupos.name.fraser.neil.plaintext.DiffMatchPatch.Operation;
- import java.util.LinkedList;

File: common/DiffMatchPatch.java [hunk 3]
```diff
@@ -31,91 +31,74 @@
...
  * Source: https://github.com/google/diff-match-patch
  * TODO: Add diff-patch-match to WORKSPACE and use directly instead of copying the code.
  */
 
-/**
- * Class containing the diff, match and patch methods.
- * Also contains the behaviour settings.
- */
+/** Class containing the diff, match and patch methods. Also contains the behaviour settings. */
 public class DiffMatchPatch {
```
Comment (val-fed) [inline common/DiffMatchPatch.java]: Looks like package name is incorrect. Is it ok?
Maybe the name should be `com.google.startupos.common`?

Comment (val-fed) [reply common/DiffMatchPatch.java]: Maybe replace `import java.util.*` to single class import?

File: common/DiffMatchPatch.java [hunk 4]
```diff
@@ -146,19 +128,19 @@ protected LinesToCharsResult(String chars1, String chars2,
...
+   * @param checklines Speedup flag. If false, then don't run a line-level diff first to identify
+   *     the changed areas. If true, then run a faster slightly less optimal diff.
+   * @param deadline Time when the diff should be complete by. Used internally for recursive calls.
+   *     Users should set DiffTimeout instead.
    * @return Linked List of Diff objects.
    */
-  private LinkedList<Diff> diff_main(String text1, String text2,
-                                     boolean checklines, long deadline) {
+  private LinkedList<Diff> diff_main(
+      String text1, String text2, boolean checklines, long deadline) {
```
Comment (val-fed) [inline common/DiffMatchPatch.java]: The explicit type argument `<Diff>` (`diffs = new LinkedList<Diff>()`) can be replaced with `<>` (line 152). 
Also below.

File: common/TextDifferencer.java (hunk 1 above)
Comment (oferb) [reply common/TextDifferencer.java]: Done

File: common/tests/TextDifferencerTool.java (hunk 2 above)
Comment (oferb) [reply common/tests/TextDifferencerTool.java]: Done

File: common/DiffMatchPatch.java (hunk 3 above)
Comment (oferb) [reply common/DiffMatchPatch.java]: As noted, this code is taken as-is from another repo.

**Pull request merged**